### PR TITLE
Allow filtering by last_active_time on the recent coach report endpoint.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Update the recent coach report API endpoint to actually only return recent results.

## TODO

- [ ] Have tests been written for the new code?

### CODE REVIEW

File: kolibri/plugins/coach/api.py [hunk 1]
```diff
@@ -80,5 +85,14 @@ class RecentReportViewSet(viewsets.ModelViewSet):
 
     def get_queryset(self):
         query_node = ContentNode.objects.get(pk=self.kwargs['content_node_id'])
-        recent_content_items = ContentSummaryLog.objects.filter_by_topic(query_node).order_by('end_timestamp').values_list('content_id')
+        if self.request.query_params.get('last_active_time'):
+            # Last active time specified
+            datetime_cutoff = parse(self.request.query_params.get('last_active_time'))
+        else:
+            datetime_cutoff = timezone.now() - datetime.timedelta(7)
```
Comment (DXCanas) [inline kolibri/plugins/coach/api.py:92]: Ah cool, so if I don't pass it, it'll as for 7 days of reports by default? What was it you were saying about the servers not being reliable with their time logs?

File: kolibri/plugins/coach/serializers.py [hunk 2]
```diff
@@ -95,12 +99,15 @@ def get_progress(self, target_node):
...
         else:
             # filter logs by a specific leaf node and compute stats over queryset
-            leaf_node_stats = ContentSummaryLog.objects \
+            leaf_node_stats_query = ContentSummaryLog.objects \
                 .filter(content_id=target_node.content_id) \
-                .filter(user__in=get_members_or_user(kwargs['collection_kind'], kwargs['collection_id'])) \
-                .aggregate(total_progress=Coalesce(Sum('progress'), V(0)),
-                           log_count_total=Coalesce(Count('pk'), V(0)),
-                           log_count_complete=Coalesce(Sum(Case(When(progress=1, then=1), default=0, output_field=IntegerField())), V(0)))
+                .filter(user__in=get_members_or_user(kwargs['collection_kind'], kwargs['collection_id']))
```
Comment (DXCanas) [inline kolibri/plugins/coach/serializers.py:104]: All of these default parameters tell me that these values are optional in the query; where can I look to find what parameters are required, if there are any?

Comment (rtibbles) [reply kolibri/plugins/coach/serializers.py:104]: Check the RecentReportResource - the 'resource_ids' are the required arguments (and must be passed first in an object to `getCollection` and second to `getModel` - after the `id`).

Comment (rtibbles) [reply kolibri/plugins/coach/serializers.py:104]: Everything else is getParams.

**Comment by indirectlylit** [comment]
@jamalex @aronasorman would one of you be able to test/review this?

**Pull request merged**